src/android/CodePush.java

```diff
@@ -270,6 +270,44 @@ private void cleanOldPackageSilently() {
         }
     }
 
+    private void clearDeploymentsIfBinaryUpdated() {
+        /* check if we have a deployed package already */
+        CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+        if (deployedPackageMetadata != null) {
+            String deployedPackageTimeStamp = deployedPackageMetadata.nativeBuildTime;
+            long nativeBuildTime = Utilities.getApkEntryBuildTime(RESOURCES_BUNDLE, this.cordova.getActivity());
+            if (nativeBuildTime != -1) {
+                String currentAppTimeStamp = String.valueOf(nativeBuildTime);
+                if ((deployedPackageTimeStamp != null) && (currentAppTimeStamp != null)) {
+                    if (!deployedPackageTimeStamp.equals(currentAppTimeStamp)) {
+                        this.codePushPackageManager.cleanDeployments();
+                        this.codePushPackageManager.clearFailedUpdates();
+                        this.codePushPackageManager.clearPendingInstall();
+                        this.codePushPackageManager.clearInstallNeedsConfirmation();
+                        this.codePushPackageManager.clearBinaryFirstRunFlag();
+                    }
+                }
+            }
+        }
+    }
+
+    private void navigateToLocalDeploymentIfExists() {
+        CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+        if (deployedPackageMetadata != null) {
```

Comment:
you can compress these `if`s with
`if (deployedPackageMetadata != null && deployedPackageMetadata.localPath != null)`